hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/storage/StorageConfiguration.java

```diff
@@ -0,0 +1,167 @@
+import pl.allegro.tech.hermes.management.domain.blacklist.TopicBlacklistRepository;
+import pl.allegro.tech.hermes.management.domain.dc.MultiDcRepositoryCommandExecutor;
+import pl.allegro.tech.hermes.management.infrastructure.blacklist.ZookeeperTopicBlacklistRepository;
+import pl.allegro.tech.hermes.management.infrastructure.dc.DcNameProvider;
+import pl.allegro.tech.hermes.management.infrastructure.dc.DcNameSource;
+import pl.allegro.tech.hermes.management.infrastructure.dc.DefaultDcNameProvider;
+import pl.allegro.tech.hermes.management.infrastructure.dc.EnvironmentVariableDcNameProvider;
+import pl.allegro.tech.hermes.management.infrastructure.zookeeper.ZookeeperClient;
+import pl.allegro.tech.hermes.management.infrastructure.zookeeper.ZookeeperClientManager;
+import pl.allegro.tech.hermes.management.infrastructure.zookeeper.ZookeeperRepositoryManager;
+
+import javax.annotation.PostConstruct;
+
+@Configuration
+@EnableConfigurationProperties(StorageClustersProperties.class)
+public class StorageConfiguration {
+
+    @Autowired
+    StorageClustersProperties storageClustersProperties;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Bean
+    DcNameProvider dcNameProvider() {
+        if(storageClustersProperties.getDcNameSource() == DcNameSource.ENV) {
+            return new EnvironmentVariableDcNameProvider(storageClustersProperties.getDcNameSourceEnv());
+        } else {
+            return new DefaultDcNameProvider();
+        }
+    }
+
+    @Bean(initMethod = "start", destroyMethod = "stop")
+    ZookeeperClientManager clientManager() {
+        return new ZookeeperClientManager(storageClustersProperties, dcNameProvider());
+    }
+
+    @Bean(initMethod = "start")
+    ZookeeperRepositoryManager repositoryManager() {
+        return new ZookeeperRepositoryManager(clientManager(), dcNameProvider(), objectMapper,
+                zookeeperPaths());
+    }
+
+    @Bean
+    ZookeeperPaths zookeeperPaths() {
+        return new ZookeeperPaths(storageClustersProperties.getPathPrefix());
+    }
+
+    @Bean
+    MultiDcRepositoryCommandExecutor multiDcRepositoryCommandExecutor() {
+        return new MultiDcRepositoryCommandExecutor(repositoryManager(), storageClustersProperties.isTransactional());
+    }
+
+    @Bean
+    public SharedCounter sharedCounter() {
+        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
+        return new SharedCounter(curatorFramework,
+                storageClustersProperties.getSharedCountersExpiration(),
+                storageClustersProperties.getRetrySleep(),
+                storageClustersProperties.getRetryTimes());
+    }
+
+    @Bean
+    public DistributedEphemeralCounter distributedCounter() {
+        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
+        return new DistributedEphemeralCounter(curatorFramework);
+    }
+
+    @Bean
+    GroupRepository groupRepository() {
```

Comment:
These below repositories are misleading now as they still allow for making changes in local DC. Because of this, a developer instead of using *Command approach and making a change in all DCs, can by mistake make only change in single DC. I’m wondering how to solve this. Maybe we should remove write operation from local repositories or at least add 'local' prefix to the repository names.

Reply:
Discussed face to face, we're going to postpone this, currently it isn't a blocker, rather than 'nice to have' feature.